src/main/java/tutorease/address/model/LessonSchedule.java

```diff
@@ -53,16 +55,18 @@ public boolean hasLesson(Lesson lesson) {
      * @throws NullPointerException If the specified lesson is null.
      */
     public void addLesson(Lesson lesson) {
+        requireNonNull(lesson);
         lessons.add(lesson);
     }
 
     /**
-     * Deletes the lesson at the specified index from the lesson list.
+     * Deletes the specified lesson from the lesson list.
      *
-     * @param index The index of the lesson to be removed. Must be a valid index.
+     * @param lesson The lesson to be removed. Must exist in the lesson list.
      */
-    public void deleteLesson(int index) {
-        lessons.remove(index);
+    public void deleteLesson(Lesson lesson) {
```

Comment:
Instead of index, we are using the lesson itself...It is okay because the delete person in TutorEase is also following this way?

Reply:
Yes it is essential for deleting correctly